### ISSUE TITLE
Use ipaddress for IP address parsing?

### ISSUE THREAD

**Issue by waveform80** [tjguk/networkzero#8]
Just run across this project - looks excellent so far, despite early days! Trawling briefly through the core module I noticed a few things (IP address testing, parsing, etc.) which could probably be handled a bit more easily/elegantly with Python's built-in [ipaddress module](https://docs.python.org/3/library/ipaddress.html).

The major issue with this module is that it was introduced in 3.3, so it's not available on 3.2 (an issue if you're intending to support Raspbian Wheezy) or 2.7 (which I note you explicitly support in setup.py). I've tackled this in a couple of prior projects by including a dependency on `ipaddr` which is the PyPI module that 3.3's ipaddress module originated from) - here's an example of what I did in [compoundpi's dependencies](https://github.com/waveform80/compoundpi/blob/master/compoundpi/__init__.py#L48) and a [brief monkey-patch](https://github.com/waveform80/compoundpi/blob/master/compoundpi/ipaddress.py) module I wrote so I could use it equivalently in Py 2.x and 3.x.

Anyway, just a thought in case you weren't aware of the module - I'd understand entirely if you wanted to avoid the whole dependency quagmire!

**Comment by tjguk** [comment]
Thanks for looking. Yes: I initially thought of using ipaddress before I 
decided that 2.7 support was desirable. I haven't actually settled on a 
lowest version for 3.x. As it happens I'm developing against 3.4 so 
there's that, but I'm aware that Raspbian pre-Jessie was 3.2 so I might 
well want to go back that far.

Initially I'd been trying to keep dependencies down. (And @ntoll would 
like them down even further). But I recently added netifaces as an 
external dependency when there seemed to be no cross-platform built-in 
for determining possible IP4 addresses.

In short: if I find insurmountable issues with ip parsing etc. I'll 
almost certainly turn to ipaddress and/or its backport. If not, though, 
I'll stay builtin.

**Comment by waveform80** [comment]
Sounds like a good plan to me!
